docs/src/components/TableCode.tsx

```diff
@@ -18,23 +18,30 @@ export function TableCode({ children, printWidth = 40, ...props }: TableCodeProp
     let groupIndex = 0;
     parts.forEach((child, index) => {
       if (index === 0) {
-        unionGroups.push([child]);
-        return;
+        unionGroups.push([]);
       }
 
-      if (getChildrenText(child).trim() === '|' && depth < 1) {
-        groupIndex += 1;
+      const str = getChildrenText(child);
+
+      // Solo function return values shouldn't be broken up, e.g. in:
+      // `(value) => string | string[] | null | Promise`
+      if (str.includes('=>') && depth < 1) {
+        depth += 1;
+      }
```

Comment:
before (incorrect, shows `validate` prop as a union type that includes a function):

<img width="813" alt="image" src="https://github.com/user-attachments/assets/703ba4d1-ae40-44e2-953f-822a46e524d5" />


after (correct, `validate` prop is a function that returns a union type):

<img width="808" alt="image" src="https://github.com/user-attachments/assets/4e01f102-85d4-4deb-9ef1-c253b7fe2051" />